lib/src/main/java/com/auth0/jwt/impl/PayloadSerializer.java

```diff
@@ -21,37 +20,47 @@ private PayloadSerializer(Class<ClaimsHolder> t) {
 
     @Override
     public void serialize(ClaimsHolder holder, JsonGenerator gen, SerializerProvider provider) throws IOException {
-        HashMap<Object, Object> safePayload = new HashMap<>();
+        
+        gen.writeStartObject();
         for (Map.Entry<String, Object> e : holder.getClaims().entrySet()) {
             switch (e.getKey()) {
                 case PublicClaims.AUDIENCE:
                     if (e.getValue() instanceof String) {
-                        safePayload.put(e.getKey(), e.getValue());
+                        gen.writeFieldName(e.getKey());
+                        gen.writeString((String)e.getValue());
                         break;
                     }
                     String[] audArray = (String[]) e.getValue();
                     if (audArray.length == 1) {
-                        safePayload.put(e.getKey(), audArray[0]);
+                        gen.writeFieldName(e.getKey());
+                        gen.writeString(audArray[0]);
                     } else if (audArray.length > 1) {
-                        safePayload.put(e.getKey(), audArray);
+                        gen.writeFieldName(e.getKey());
+                        gen.writeStartArray();
+                        for(String aud : audArray) {
+                            gen.writeString(aud);
+                        }
+                        gen.writeEndArray();
                     }
                     break;
                 case PublicClaims.EXPIRES_AT:
                 case PublicClaims.ISSUED_AT:
                 case PublicClaims.NOT_BEFORE:
-                    safePayload.put(e.getKey(), dateToSeconds((Date) e.getValue()));
+                    gen.writeFieldName(e.getKey());
+                    gen.writeNumber(dateToSeconds((Date) e.getValue()));
                     break;
                 default:
+                    gen.writeFieldName(e.getKey());
                     if (e.getValue() instanceof Date) {
```

Comment:
this `default` case could be merged with the 3 known claims above (which are dates as well), or rephrasing, this could be deleted:

```java
case PublicClaims.EXPIRES_AT:
case PublicClaims.ISSUED_AT:
case PublicClaims.NOT_BEFORE:
   gen.writeFieldName(e.getKey());
   gen.writeNumber(dateToSeconds((Date) e.getValue()));
   break;
```

Reply:
Deleted according to your comment.